### PR TITLE
Fix GCE deployment flow

### LINKED ISSUE #2056: GCE credentials should only be using uploading .json
The attributes part is broken anyway. Make json upload default and add handler for uploads.

### PR DESCRIPTION
**Pull request by hatched**
Added JSON file upload support to the GCE provider deployment flow.
Fixed the formatting of the private key for the GCE provider deployment flow.
Updated the ExpandingRow component to support dynamic height content.
Fixed the font line-height and width on the InsetSelect component.

Fixes #2056

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5910/
Test PASSed.

**Comment by bac** [comment]
QA:

Bootstrapped on AWS using:

```
juju bootstrap aws aws --config identity-url=https://api.jujucharms.com/identity
juju grant everyone@external addmodel
```

Logged in via USSO.  Tried to add an AWS credential and got JS error:

```
Uncaught TypeError: Cannot read property 'map' of undefined
_handleAddCredentials
```

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5911/
Test PASSed.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js [hunk 1]
```diff
@@ -165,16 +167,11 @@ describe('DeploymentCredentialAdd', function() {
...
-                key="private-key"
-                label="Client secret"
-                required={true}
-                ref="private-key"
-                validate={[{
-                  regex: /\S+/,
-                  error: 'This field is required.'
-                }]} />,
+                key={"private-key"}
+                option={{key: 'Client secret'}}
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js]: Why not 'Private key'?

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js [hunk 2]
```diff
@@ -50,7 +50,9 @@ describe('DeploymentCredentialAdd', function() {
             title: 'Client e-mail address'
           }, {
             id: 'private-key',
-            title: 'Client secret'
+            title: 'Client secret',
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js]: Private key?

File: jujugui/static/gui/src/app/components/file-field/file-field.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/file-field/file-field.js]: s/environments/models

File: jujugui/static/gui/src/app/components/file-field/test-file-field.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
```
Comment (urosj) [inline jujugui/static/gui/src/app/components/file-field/test-file-field.js]: s/environments/models

Comment (urosj) [inline jujugui/static/gui/src/app/components/file-field/test-file-field.js]: and repo is on github.

File: jujugui/static/gui/src/app/components/file-field/file-field.js (hunk 3 above)
Comment (urosj) [inline jujugui/static/gui/src/app/components/file-field/file-field.js]: Repo is on github as well.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js]: This was actually supposed to have been changed, must have missed it, oops, fixed!

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/test-add.js]: fixed

File: jujugui/static/gui/src/app/components/file-field/file-field.js (hunk 3 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/file-field/file-field.js]: fixed

File: jujugui/static/gui/src/app/components/file-field/test-file-field.js (hunk 4 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/file-field/test-file-field.js]: fixed

**Comment by hatched** [comment]
Thanks all!

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5912/
Test PASSed.

**Pull request merged**